In [139]:
import os
import pandas as pd

Read in the data

In [164]:
data = pd.read_excel('/Users/elizabeth 1/Library/CloudStorage/Box-Box/Wu Lab/Project - 2024/Functional screening/240318_Natalie_Moe requested genes Query Sigma 226 for 162.xlsx')
data

DNA plate well     OD  Quadrant/DNA plate #  Vol of DNA for 20 ng/ul  \
0               A1  0.539                   1.0                12.334962   
1              A11  0.490                   1.0                11.293759   
2              A13  0.533                   1.0                13.215395   
3              A15  0.513                   1.0                14.655386   
4              A17  0.468                   1.0                12.892080   
..             ...    ...                   ...                      ...   
221            NaN    NaN                   NaN                      NaN   
222            NaN    NaN                   NaN                      NaN   
223            NaN    NaN                   NaN                      NaN   
224            NaN    NaN                   NaN                      NaN   
225            NaN    NaN                   NaN                      NaN   

    Unnamed: 4  Unnamed: 5 Unnamed: 6  NCBI Gene ID             Ensembl  \
0       C-1F05           1   BAA-3C03         29811  ENSMUSG00000004558   
1       C-1O07           1   BAA-1H04         56791  ENSMUSG00000027078   
2      C-21G18          21  BAA-82D09         13390  ENSMUSG00000041911   
3       C-1P01           1   BAA-3H01        231858  ENSMUSG00000029576   
4      C-21H13          21  BAA-83D07         20322  ENSMUSG00000027227   
..         ...         ...        ...           ...                 ...   
221    C-46H24          46  BAB-85D12         73826  ENSMUSG00000041815   
222    C-10A20          10  BAA-38A10         19132  ENSMUSG00000023484   
223    C-37D24          37  BAB-49B12         19713  ENSMUSG00000030110   
224    C-27F15          27   BAB-8C08         22348  ENSMUSG00000037771   
225     C-4A08           4  BAA-14A04         22042  ENSMUSG00000022797   

    Moe requested genes_Name  ...  \
0                      Ndrg2  ...   
1                     Ube2l6  ...   
2                       Dlx1  ...   
3                      Radil  ...   
4                       sord  ...   
..                       ...  ...   
221                  poldip3  ...   
222                     Prph  ...   
223                      Ret  ...   
224                  Slc32a1  ...   
225                     tfrc  ...   

                                               5_Flank  \
0    tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
1    tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
2    tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
3    tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
4    tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
..                                                 ...   
221  tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
222  tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
223  tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
224  tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   
225  tcttccatttcaggtgtcgtgaggctagcatcgattgatcaacaag...   

                                               3_Flank  Annotation_ID  \
0    TGCCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...           6646   
1    TACCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...          11372   
2    TGCCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...           2782   
3    TACCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...           8500   
4    TGCCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...           9894   
..                                                 ...            ...   
221  TACCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...           7929   
222  TGCCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...           8191   
223  CCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCCCTA...           8666   
224  TACCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...           9600   
225  TGCCCAACTTTCTTGTACAAAGTGGTTGATATCGGTAAGCCTATCC...          10584   

       query  entrezgene                Sigma Curated Annotated_name  \
0      NDRG2       57447                        NDRG family member 2   
1     UBE2L6        9246   

In [165]:
data = data.loc[:,['DNA plate well', 'Quadrant/DNA plate #', 'Row', 'Col', 'Moe requested genes_Name']]
plate_well = data['DNA plate well']

In [123]:
# check if there's duplicates in the first column
wells = []
for well in plate_well:
    if well not in wells:
        wells.append(well)
    else:
        print(well)

nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan
nan


In [166]:
data.drop(columns=['Row', 'Col'], inplace=True)
data

DNA plate well  Quadrant/DNA plate # Moe requested genes_Name
0               A1                   1.0                    Ndrg2
1              A11                   1.0                   Ube2l6
2              A13                   1.0                     Dlx1
3              A15                   1.0                    Radil
4              A17                   1.0                     sord
..             ...                   ...                      ...
221            NaN                   NaN                  poldip3
222            NaN                   NaN                     Prph
223            NaN                   NaN                      Ret
224            NaN                   NaN                  Slc32a1
225            NaN                   NaN                     tfrc

[226 rows x 3 columns]

In [167]:
data.columns = ['well', 'plate', 'gene']
data

well  plate     gene
0     A1    1.0    Ndrg2
1    A11    1.0   Ube2l6
2    A13    1.0     Dlx1
3    A15    1.0    Radil
4    A17    1.0     sord
..   ...    ...      ...
221  NaN    NaN  poldip3
222  NaN    NaN     Prph
223  NaN    NaN      Ret
224  NaN    NaN  Slc32a1
225  NaN    NaN     tfrc

[226 rows x 3 columns]

In [136]:
print(len(data))
# Use a Boolean condition directly to filter out rows where 'A' is not equal to 2
data = data[(data['plate'] == 1.0) | (data['plate'] == 2.0)]       
print(len(data))     

226
192


Create the Plate Map Dataframes

In [144]:
cols = [i for i in range(1,25)] # half-open range (start @ start, up to but not including stop) -- also used for list slicing & iteration/loops
rows = [chr(i) for i in range(ord('A'), ord('H')+1)] # list comprehension
# ord for ordinal --> numeric value assigned to a character, returns char's ASCII val. Go up to H+1 b/c of half-open range (any string char has an ASCII val)
# chr(i) returns character that corresponds to the number, the numbers 'i' being the ASCII values from A-I

In [178]:
# create empty dataframes
plate_1 = pd.DataFrame(index=rows, columns=cols)
plate_2 = pd.DataFrame(index=rows, columns=cols) # can't se it equal to plate_1 bc plate_2 will be a variable pointing to plate_1, so any changes in plate_2 will be reflected in plate_1 and vice versa
plate_1.index.name = 'Plate 1'
plate_2.index.name = 'Plate 2'
plate_1

1    2    3    4    5    6    7    8    9    10  ...   15   16   17  \
Plate 1                                                    ...                  
A        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   
B        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   
C        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   
D        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   
E        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   
F        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   
G        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   
H        NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  ...  NaN  NaN  NaN   

          18   19   20   21   22   23   24  
Plate 1                                     
A        NaN  NaN  NaN  NaN  NaN  NaN  NaN  
B        NaN  NaN  NaN  NaN  NaN  NaN  NaN  
C        NaN  NaN  NaN  NaN  NaN  NaN  NaN  
D        NaN  NaN  NaN  NaN  NaN  NaN  NaN  
E        NaN  NaN  NaN  NaN  NaN  NaN  NaN  
F        NaN  NaN  NaN  NaN  NaN  NaN  NaN  
G        NaN  NaN  NaN  NaN  NaN  NaN  NaN  
H        NaN  NaN  NaN  NaN  NaN  NaN  NaN  

[8 rows x 24 columns]

In [177]:
data_1

well  plate      gene
0    A1    1.0     Ndrg2
1   A11    1.0    Ube2l6
2   A13    1.0      Dlx1
3   A15    1.0     Radil
4   A17    1.0      sord
..  ...    ...       ...
91  O23    1.0  Calcoco2
92   O3    1.0      Mdfi
93   O5    1.0      myo6
94   O7    1.0    Gpr157
95   O9    1.0    Higd1a

[96 rows x 3 columns]

In [179]:
for index, row in data_1.iterrows():
    plate = row['plate']
    gene = row['gene']   
    well = row['well']
    plate_row = well[0]
    plate_col = int(well[1:])
    
    plate_1.loc[plate_row,plate_col] = gene # use loc instead of iloc to access by index name instead of #
    print(plate_row, plate_col, gene)
        
plate_1
        

A 1 Ndrg2
A 11 Ube2l6
A 13 Dlx1
A 15 Radil
A 17 sord
A 19 vim
A 21 Cd68
A 23 Trim62
A 3 Irf5
A 5 Ndrg2
A 7 Foxa3
A 9 Gsta3
C 1 Gsta3
C 11 Krt19
C 13 Smarcd1
C 15 Npr2
C 17 Npr2
C 19 Sgk2
C 21 Npr2
C 23 Npr2
C 3 Reep6
C 5 Aig1
C 7 Glo1
C 9 Gfer
E 1 Fes
E 11 Acss3
E 13 Igfbp2
E 15 Ccdc85c
E 17 Cib2
E 19 Gpr157
E 21 Cib2
E 23 Dpf1
E 3 Epha4
E 5 Osr2
E 7 Amhr2
E 9 Ret
G 1 Celf5
G 11 Hsd17b1
G 13 Lurap1l
G 15 Tmem121
G 17 Fbxl19
G 19 Sema4d
G 21 Igf2
G 23 Lcn2
G 3 Tex101
G 5 Wnt4
G 7 Slc9a3r2
G 9 Bcam
I 1 Dpf3
I 11 Slc29a4
I 13 Tcf7
I 15 Esrrb
I 17 Grin1
I 19 vim
I 21 Gpr157
I 23 Esrrb
I 3 Trim62
I 5 Nxph4
I 7 Bin2
I 9 Alcam
K 1 Npr3
K 11 Ldhd
K 13 Slc2a3
K 15 Rassf2
K 17 Mypop
K 19 Batf3
K 21 Fbxo10
K 23 Lhx6
K 3 Gpr157
K 5 Gpr157
K 7 Higd1a
K 9 Ptpn18
M 1 Txndc16
M 11 Eps8l2
M 13 Clmn
M 15 Higd1a
M 17 Angel1
M 19 B4galt3
M 21 Apln
M 23 Tead2
M 3 Sp6
M 5 Enpep
M 7 Rpia
M 9 Rasgrp4
O 1 Spata2l
O 11 Grin1
O 13 Npr3
O 15 Rab11fip4
O 17 Fbxo17
O 19 Spata2l
O 21 Tubg2
O 23 Calcoco2
O 3 Mdfi
O 5

1    2       3    4       5    6         7    8        9    10  \
Plate 1                                                                        
A          Ndrg2  NaN    Irf5  NaN   Ndrg2  NaN     Foxa3  NaN    Gsta3  NaN   
B            NaN  NaN     NaN  NaN     NaN  NaN       NaN  NaN      NaN  NaN   
C          Gsta3  NaN   Reep6  NaN    Aig1  NaN      Glo1  NaN     Gfer  NaN   
D            NaN  NaN     NaN  NaN     NaN  NaN       NaN  NaN      NaN  NaN   
E            Fes  NaN   Epha4  NaN    Osr2  NaN     Amhr2  NaN      Ret  NaN   
F            NaN  NaN     NaN  NaN     NaN  NaN       NaN  NaN      NaN  NaN   
G          Celf5  NaN  Tex101  NaN    Wnt4  NaN  Slc9a3r2  NaN     Bcam  NaN   
H            NaN  NaN     NaN  NaN     NaN  NaN       NaN  NaN      NaN  NaN   
I           Dpf3  NaN  Trim62  NaN   Nxph4  NaN      Bin2  NaN    Alcam  NaN   
K           Npr3  NaN  Gpr157  NaN  Gpr157  NaN    Higd1a  NaN   Ptpn18  NaN   
M        Txndc16  NaN     Sp6  NaN   Enpep  NaN      Rpia  NaN  Rasgrp4  NaN   
O        Spata2l  NaN    Mdfi  NaN    myo6  NaN    Gpr157  NaN   Higd1a  NaN   

         ...         15   16      17   18       19   20      21   22  \
Plate 1  ...                                                           
A        ...      Radil  NaN    sord  NaN      vim  NaN    Cd68  NaN   
B        ...        NaN  NaN     NaN  NaN      NaN  NaN     NaN  NaN   
C        ...       Npr2  NaN    Npr2  NaN     Sgk2  NaN    Npr2  NaN   
D        ...        NaN  NaN     NaN  NaN      NaN  NaN     NaN  NaN   
E        ...    Ccdc85c  NaN    Cib2  NaN   Gpr157  NaN    Cib2  NaN   
F        ...        NaN  NaN     NaN  NaN      NaN  NaN     NaN  NaN   
G        ...    Tmem121  NaN  Fbxl19  NaN   Sema4d  NaN    Igf2  NaN   
H        ...        NaN  NaN     NaN  NaN      NaN  NaN     NaN  NaN   
I        ...      Esrrb  NaN   Grin1  NaN      vim  NaN  Gpr157  NaN   
K        ...     Rassf2  NaN   Mypop  NaN    Batf3  NaN  Fbxo10  NaN   
M        ...     Higd1a  NaN  Angel1  NaN  B4galt3  NaN    Apln  NaN   
O        ...  Rab11fip4  NaN  Fbxo17  NaN  Spata2l  NaN   Tubg2  NaN   

               23   24  
Plate 1                 
A          Trim62  NaN  
B             NaN  NaN  
C            Npr2  NaN  
D             NaN  NaN  
E            Dpf1  NaN  
F             NaN  NaN  
G            Lcn2  NaN  
H             NaN  NaN  
I           Esrrb  NaN  
K            Lhx6  NaN  
M           Tead2  NaN  
O        Calcoco2  NaN  

[12 rows x 24 columns]

In [180]:
for index, row in data_2.iterrows():
    plate = row['plate']
    gene = row['gene']
    well = row['well']
    plate_row = well[0]
    plate_col = int(well[1:])

    plate_2.loc[plate_row,plate_col] = gene # use loc instead of iloc to access by index name instead of #
    print(plate_row, plate_col, gene)

plate_2

A 10 Fmod
A 12 Ptpn6
A 14 Plvap
A 16 Vill
A 18 Slc16a14
A 2 Avil
A 20 Ndufa4l2
A 22 Mid1
A 24 Fbxl14
A 4 Plekhg6
A 6 Nr5a1
A 8 Slc25a23
C 10 prdx2
C 12 Cryaa
C 14 Gng2
C 16 Cd164l2
C 18 Tekt2
C 2 Thap11
C 20 Agmo
C 22 Npr3
C 24 Rtn4rl1
C 4 Npr3
C 6 Bmp7
C 8 Vps4a
E 10 Dnajc15
E 12 Ckb
E 14 Vax2
E 16 Mpzl1
E 18 Brsk1
E 2 Ckb
E 20 Sult1c2
E 22 Uck2
E 24 Nt5e
E 4 Mmp2
E 6 Ap1m2
E 8 Amhr2
G 10 Uck2
G 12 Rrad
G 14 Pfkl
G 16 Slc9a3r2
G 18 Mlkl
G 2 Pim2
G 20 Tacstd2
G 22 Ret
G 24 Vill
G 4 Mpzl1
G 6 Pim2
G 8 Ggh
I 10 Ltk
I 12 Dact3
I 14 Npr2
I 16 Npr2
I 18 Dpysl5
I 2 Pfkl
I 20 Uck2
I 22 Ret
I 24 Vps26a
I 4 Ccdc92
I 6 Epha4
I 8 Diras2
K 10 Sgk2
K 12 Tbx15
K 14 Pfkl
K 16 Uck2
K 18 Sgk2
K 2 Pim2
K 20 sf3b2
K 22 Sgk2
K 24 Ret
K 4 Podnl1
K 6 Amhr2
K 8 Rspo1
M 10 Pgk1
M 12 Bcl2l11
M 14 Prr5
M 16 St6gal1
M 18 Ptpn18
M 2 Ret
M 20 Gapdh
M 22 Avil
M 24 Nup50
M 4 Crabp1
M 6 Pgk1
M 8 Pfkl
O 10 Tiam1
O 12 Arsi
O 14 Ostn
O 16 dbn1
O 18 Rims3
O 2 Spryd7
O 20 mcm6
O 22 B4galt3
O 24 Ccdc136
O 4 Slc29a4
O 6 Ser

1       2    3        4    5         6    7         8    9   \
Plate 2                                                                 
A        NaN    Avil  NaN  Plekhg6  NaN     Nr5a1  NaN  Slc25a23  NaN   
B        NaN     NaN  NaN      NaN  NaN       NaN  NaN       NaN  NaN   
C        NaN  Thap11  NaN     Npr3  NaN      Bmp7  NaN     Vps4a  NaN   
D        NaN     NaN  NaN      NaN  NaN       NaN  NaN       NaN  NaN   
E        NaN     Ckb  NaN     Mmp2  NaN     Ap1m2  NaN     Amhr2  NaN   
F        NaN     NaN  NaN      NaN  NaN       NaN  NaN       NaN  NaN   
G        NaN    Pim2  NaN    Mpzl1  NaN      Pim2  NaN       Ggh  NaN   
H        NaN     NaN  NaN      NaN  NaN       NaN  NaN       NaN  NaN   
I        NaN    Pfkl  NaN   Ccdc92  NaN     Epha4  NaN    Diras2  NaN   
K        NaN    Pim2  NaN   Podnl1  NaN     Amhr2  NaN     Rspo1  NaN   
M        NaN     Ret  NaN   Crabp1  NaN      Pgk1  NaN      Pfkl  NaN   
O        NaN  Spryd7  NaN  Slc29a4  NaN  Serping1  NaN      Nt5e  NaN   

              10  ...   15        16   17        18   19        20   21  \
Plate 2           ...                                                     
A           Fmod  ...  NaN      Vill  NaN  Slc16a14  NaN  Ndufa4l2  NaN   
B            NaN  ...  NaN       NaN  NaN       NaN  NaN       NaN  NaN   
C          prdx2  ...  NaN   Cd164l2  NaN     Tekt2  NaN      Agmo  NaN   
D            NaN  ...  NaN       NaN  NaN       NaN  NaN       NaN  NaN   
E        Dnajc15  ...  NaN     Mpzl1  NaN     Brsk1  NaN   Sult1c2  NaN   
F            NaN  ...  NaN       NaN  NaN       NaN  NaN       NaN  NaN   
G           Uck2  ...  NaN  Slc9a3r2  NaN      Mlkl  NaN   Tacstd2  NaN   
H            NaN  ...  NaN       NaN  NaN       NaN  NaN       NaN  NaN   
I            Ltk  ...  NaN      Npr2  NaN    Dpysl5  NaN      Uck2  NaN   
K           Sgk2  ...  NaN      Uck2  NaN      Sgk2  NaN     sf3b2  NaN   
M           Pgk1  ...  NaN   St6gal1  NaN    Ptpn18  NaN     Gapdh  NaN   
O          Tiam1  ...  NaN      dbn1  NaN     Rims3  NaN      mcm6  NaN   

              22   23       24  
Plate 2                         
A           Mid1  NaN   Fbxl14  
B            NaN  NaN      NaN  
C           Npr3  NaN  Rtn4rl1  
D            NaN  NaN      NaN  
E           Uck2  NaN     Nt5e  
F            NaN  NaN      NaN  
G            Ret  NaN     Vill  
H            NaN  NaN      NaN  
I            Ret  NaN   Vps26a  
K           Sgk2  NaN      Ret  
M           Avil  NaN    Nup50  
O        B4galt3  NaN  Ccdc136  

[12 rows x 24 columns]

In [188]:
print(plate_1.isnull().sum().sum())
print(plate_2.isnull().sum().sum())
print(plate_1.notnull().sum().sum())


192
192
96


In [191]:
plate_1.iloc[1,:].notnull().sum()

0

In [193]:
# drop NA rows/cols from both plate dataframes
# Drop all rows that have only NaN values in place
plate_1.dropna(axis=0, how='all', inplace=True)
plate_1.dropna(axis=1, how='all', inplace=True)
plate_1

1       3       5         7        9        11       13  \
Plate 1                                                                 
A          Ndrg2    Irf5   Ndrg2     Foxa3    Gsta3   Ube2l6     Dlx1   
C          Gsta3   Reep6    Aig1      Glo1     Gfer    Krt19  Smarcd1   
E            Fes   Epha4    Osr2     Amhr2      Ret    Acss3   Igfbp2   
G          Celf5  Tex101    Wnt4  Slc9a3r2     Bcam  Hsd17b1  Lurap1l   
I           Dpf3  Trim62   Nxph4      Bin2    Alcam  Slc29a4     Tcf7   
K           Npr3  Gpr157  Gpr157    Higd1a   Ptpn18     Ldhd   Slc2a3   
M        Txndc16     Sp6   Enpep      Rpia  Rasgrp4   Eps8l2     Clmn   
O        Spata2l    Mdfi    myo6    Gpr157   Higd1a    Grin1     Npr3   

                15      17       19      21        23  
Plate 1                                                
A            Radil    sord      vim    Cd68    Trim62  
C             Npr2    Npr2     Sgk2    Npr2      Npr2  
E          Ccdc85c    Cib2   Gpr157    Cib2      Dpf1  
G          Tmem121  Fbxl19   Sema4d    Igf2      Lcn2  
I            Esrrb   Grin1      vim  Gpr157     Esrrb  
K           Rassf2   Mypop    Batf3  Fbxo10      Lhx6  
M           Higd1a  Angel1  B4galt3    Apln     Tead2  
O        Rab11fip4  Fbxo17  Spata2l   Tubg2  Calcoco2

In [194]:
plate_2.dropna(axis=0, how='all', inplace=True)
plate_2.dropna(axis=1, how='all', inplace=True)
plate_2

2        4         6         8        10       12     14  \
Plate 2                                                                 
A          Avil  Plekhg6     Nr5a1  Slc25a23     Fmod    Ptpn6  Plvap   
C        Thap11     Npr3      Bmp7     Vps4a    prdx2    Cryaa   Gng2   
E           Ckb     Mmp2     Ap1m2     Amhr2  Dnajc15      Ckb   Vax2   
G          Pim2    Mpzl1      Pim2       Ggh     Uck2     Rrad   Pfkl   
I          Pfkl   Ccdc92     Epha4    Diras2      Ltk    Dact3   Npr2   
K          Pim2   Podnl1     Amhr2     Rspo1     Sgk2    Tbx15   Pfkl   
M           Ret   Crabp1      Pgk1      Pfkl     Pgk1  Bcl2l11   Prr5   
O        Spryd7  Slc29a4  Serping1      Nt5e    Tiam1     Arsi   Ostn   

               16        18        20       22       24  
Plate 2                                                  
A            Vill  Slc16a14  Ndufa4l2     Mid1   Fbxl14  
C         Cd164l2     Tekt2      Agmo     Npr3  Rtn4rl1  
E           Mpzl1     Brsk1   Sult1c2     Uck2     Nt5e  
G        Slc9a3r2      Mlkl   Tacstd2      Ret     Vill  
I            Npr2    Dpysl5      Uck2      Ret   Vps26a  
K            Uck2      Sgk2     sf3b2     Sgk2      Ret  
M         St6gal1    Ptpn18     Gapdh     Avil    Nup50  
O            dbn1     Rims3      mcm6  B4galt3  Ccdc136

In [195]:
print(plate_2)

             2        4         6         8        10       12     14  \
Plate 2                                                                 
A          Avil  Plekhg6     Nr5a1  Slc25a23     Fmod    Ptpn6  Plvap   
C        Thap11     Npr3      Bmp7     Vps4a    prdx2    Cryaa   Gng2   
E           Ckb     Mmp2     Ap1m2     Amhr2  Dnajc15      Ckb   Vax2   
G          Pim2    Mpzl1      Pim2       Ggh     Uck2     Rrad   Pfkl   
I          Pfkl   Ccdc92     Epha4    Diras2      Ltk    Dact3   Npr2   
K          Pim2   Podnl1     Amhr2     Rspo1     Sgk2    Tbx15   Pfkl   
M           Ret   Crabp1      Pgk1      Pfkl     Pgk1  Bcl2l11   Prr5   
O        Spryd7  Slc29a4  Serping1      Nt5e    Tiam1     Arsi   Ostn   

               16        18        20       22       24  
Plate 2                                                  
A            Vill  Slc16a14  Ndufa4l2     Mid1   Fbxl14  
C         Cd164l2     Tekt2      Agmo     Npr3  Rtn4rl1  
E           Mpzl1     Brsk1   Sult1c2

In [196]:
# save the unlinked plate maps
os.chdir('/Users/elizabeth 1/Library/CloudStorage/Box-Box/Wu Lab/Project - statin/8. RNA-seq/Elizabeth/LW15 analysis/LW15-Functional-Screening/plate_map')
plate_1.to_html('plate_1.html')
plate_2.to_html('plate_2.html')

### Troubleshooting
check how many duplicate well instances there are

In [85]:
data

plate Row  Col     gene
0      1.0   C    3    Ndrg2
1      1.0   H    4   Ube2l6
2      1.0   D    9     Dlx1
3      1.0   H    1    Radil
4      1.0   D    7     sord
..     ...  ..  ...      ...
221    NaN   D   12  poldip3
222    NaN   A   10     Prph
223    NaN   B   12      Ret
224    NaN   C    8  Slc32a1
225    NaN   A    4     tfrc

[226 rows x 4 columns]

In [173]:
data_1 = data[data['plate'] == 1.0] 
data_2 = data[data['plate'] == 2.0]       

note:
- if there were to be a full plate map, there's 8*24 = 192 wells per plate. 
- but the data var with plates 1&2 has 226 rows (should have 384 if complete)
- and each plate 1 and plate 2 have 96 rows, respectively

In [91]:
wells = []
for index, row in data_1.iterrows():
    row_name = row['Row']
    col_name = str(row['Col'])
    well = row_name + col_name
    wells.append(well)
    

counts_dict = {}

for well in wells:
    if well not in counts_dict:
        counts_dict[well] = 1
    else:
        counts_dict[well] += 1

for well, num in counts_dict.items():
    if num >1:
        print(well, num)

C3 2
D9 3
H1 2
D7 2
H11 2
D11 2
B5 2
A11 3
D12 2
C5 3
A2 2
G7 5
F6 2
E11 2
H7 2
D4 2
E9 2
G1 2
C2 2
F8 2
F1 2
E12 2
B8 2
B6 2
F2 2


In [118]:
def duplicate_wells(plate):
    wells = {}
    for index, row in plate.iterrows():
        # print(row)
        row_name = row['Row']
        col_name = str(row['Col'])
        well = row_name + col_name
        wells[index] = well
    
    counts_dict = {} # {A1:
    
    for index, well in wells.items():
        if well in counts_dict:
            counts_dict[well].append(index)
        else:
            counts_dict[well] = []
            counts_dict[well].append(index)
    
    for well, indices in counts_dict.items():
       indices = [index + 2 for index in indices] # update by 2 to correct for the indexing
       counts_dict[well] = indices
    
    for well, indices in counts_dict.items():
        if len(indices) > 1:
            print(well, indices)
            
duplicate_wells(data_2)


F10 [101, 106, 146, 183]
D2 [102, 131, 136, 141, 163]
F4 [103, 126, 151, 158]
G11 [104, 119]
E3 [105, 169]
B6 [108, 153]
F6 [109, 129]
H5 [111, 114]
A9 [112, 145]
D3 [115, 157, 166]
D9 [116, 188]
C7 [117, 124, 134, 154]
B12 [123, 133]
D6 [125, 127]
H4 [130, 132, 182]
H9 [139, 143]
E4 [142, 191]
A8 [144, 162, 172]
A6 [152, 161]
A1 [171, 190]
E2 [173, 181]
D7 [178, 187]


In [119]:
duplicate_wells(data_1)

C3 [2, 93]
D9 [4, 61, 74]
H1 [5, 20]
D7 [6, 83]
H11 [7, 63]
D11 [8, 71]
B5 [9, 35]
A11 [10, 58, 65]
D12 [11, 80]
C5 [13, 32, 37]
A2 [15, 89]
G7 [18, 33, 60, 76, 88]
F6 [22, 95]
E11 [27, 86]
H7 [34, 72]
D4 [38, 73]
E9 [39, 56]
G1 [41, 82]
C2 [42, 52]
F8 [50, 59]
F1 [54, 64]
E12 [57, 85]
B8 [62, 94]
B6 [70, 84]
F2 [96, 97]


In [110]:
list = [1,2,3,4]
[item + 5 for item in list]

[6, 7, 8, 9]